pkg/server/http_server.go

```diff
@@ -58,6 +58,10 @@ func InitHttpServer(ctx context.Context, cfg *config.Config) {
 			queue: utils.NewTrailingDelayQueue(processRequest, cfg.RequestAggregationDelay),
 		},
 	}
+
+	// Saves the provider and engine config parameters
+	providerName = cfg.Provider
```

Comment:
we need to make validation here in set proper provider/engine in the topology request

Reply:
Fixed in last commit.